awesome_owl/static/src/todo_list/todo_list.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+
+import { Component, useState, onMounted, useRef } from "@odoo/owl";
+import { TodoItem } from "./todo_item";
+
+export class TodoList extends Component {
+    static template = "awesome_owl.todo_list";
+    static components = { TodoItem };
+
+    todos = useState([]);
+    state = useState({ count: 0 });
+
+    setup() {
+        this.inputRef = useRef("input");
+
+        onMounted(() => {
+            if (this.inputRef.el) {
+                this.inputRef.el.focus();
+            }
+        });
+    }
+
+
+    addTodo(ev) {
+        if (ev.key === 'Enter' && ev.target.value) {
+            const newTodo = {
+                id: this.state.count = this.state.count + 1,
```

Comment:
equivalent:
```suggestion
                id: this.state.count++,
```
more readable
```suggestion
                id: this.state.count,
                ...
           };
           this.state.count++;
```